### ISSUE TITLE
Z Probe not activated in some cases, head crashes into bed

### ISSUE THREAD

**Issue by ticso24** [repetier/Repetier-Firmware#839]
bool Printer::startProbing does a coordinate test, which triggered for me:
        GCode::fatalError(PSTR("Could not activate z-probe offset due to coordinate constraints - result is inaccurate!"));
        return false;
But the return value isn't being checked in the calling functions and therefor the printer continues probing with the probe not being enabled.
I don't even understand why the head relocates at all, can't it just update the current location, it is about to move to the first probe location anyways.
This results in silly head movements, locate to a save enable position, relocate head with the probe offset and then move to the probe location.
But at least the calling functions shouldn't continue with probing after the start function failed.
I also have the following in my configuration:
    "zProbeStartScript": "G0 X50 Y50 F5000\\nM340 P0 S700",
But it doesn't work with that, because it happens later, so I really have to manually send a move command before doing a G32/G33.

**Comment by repetier** [comment]
Good point with the ignored return value. Have fixed it for next update.

Changing the position actively is not an option here as it gets used for several reasons, so it needs to do it where it is. This activates the offset between nozzle and z probe position so that move is required.

In V2 firmware we have improved this a bit as we now have abstracted printer types and can compute the closest allowed position. V1 firmware simply does not have the required information for that.
